xbmc/pvr/channels/PVRChannelGroup.cpp

```diff
@@ -376,63 +376,59 @@ void CPVRChannelGroup::SortByChannelNumber(void)
 
 CPVRChannelPtr CPVRChannelGroup::GetByClient(int iUniqueChannelId, int iClientID) const
 {
+  CPVRChannelPtr retval;
   CSingleLock lock(m_critSection);
 
-  for (unsigned int ptr = 0; ptr < m_members.size(); ptr++)
+  for (std::vector<PVRChannelGroupMember>::const_iterator it = m_members.begin(); !retval && it != m_members.end(); ++it)
   {
-    PVRChannelGroupMember groupMember = m_members.at(ptr);
-    if (groupMember.channel->UniqueID() == iUniqueChannelId &&
-        groupMember.channel->ClientID() == iClientID)
-      return groupMember.channel;
+    if ((*it).channel->UniqueID() == iUniqueChannelId &&
+        (*it).channel->ClientID() == iClientID)
+      retval = (*it).channel;
```

Comment:
From side this code style this is very good  and if I'm right this is also in performance a bit better. But also @xhaggi have a bit right, is possible which is unfortunately overlooked there.
Maybe add a small comment there
```C++
#                               Note:  for becomes broken there and on other functions with this ---, 
for (std::vector<PVRChannelGroupMember>::const_iterator it = m_members.begin(); !retval && it != m_members.end(); ++it)
```
Not sure if it is to stupid to add this, but maybe it helps ;-)